src/hatti/map/utils.cljs

```diff
@@ -267,3 +269,191 @@
+                                  :stops (generate-stops
+                                          selected-id "#631400")})]]
+          :clicked [["fill-color" (clj->js
+                                   {:property "_id"
+                                    :type "categorical"
+                                    :stops (generate-stops
+                                            selected-id "#ad2300")})]]}})
+
+(defn get-style-properties
+  [style-type style-state & [selected-id stops size-stops]]
+  (-> (get-styles selected-id stops) style-type style-state))
+
+(defn set-mapboxgl-paint-property
+  "Sets maps paint properties given layer-id and list of properties to set.
+  properties should be a list of properties that contains the propery name
+  and value in a vector. e.g. [[property1 value1] [property2 value2]"
+  [map layer-id properties]
+  (doseq [[p v] properties] (.setPaintProperty map layer-id p v)))
+
+(defn register-mapboxgl-mouse-events
+  "Register map mouse events."
+  [owner map event-chan id_string style]
+  (.off map "mousemove")
+  (.off map "click")
+  (.on map "mousemove"
+       (fn [e]
+         (let [layer-id id_string
+               features
+               (.queryRenderedFeatures
+                map (.-point e) (clj->js {:layers [layer-id]}))
+               no-of-features (.-length features)
+               view-by (om/get-props owner [:map-page :view-by])
+               selected-id (om/get-props owner [:map-page :submission-clicked
+                                                :id])]
+           (set! (.-cursor (.-style (.getCanvas map)))
+                 (if (pos? (.-length features)) "pointer" ""))
+           (when-not view-by
+             (if (= no-of-features 1)
+
+               (set-mapboxgl-paint-property
+                map layer-id
+                (get-style-properties
+                 style :hover (or (-> (first features) (aget "properties")
+                                      (aget "id"))
+                                  (-> (first features) (aget "properties")
+                                      (aget _id))))))
+             (do
+               (set-mapboxgl-paint-property
+                map layer-id (get-style-properties style :normal))
+               (when selected-id
+                 (set-mapboxgl-paint-property
+                  map layer-id (get-style-properties style :clicked
+                                                     selected-id))))))))
+  (.on map "click"
+       (fn [e]
+         (let [layer-id id_string
+               features
+               (.queryRenderedFeatures
+                map (.-point e) (clj->js {:layers [layer-id]}))
+               no-of-features (.-length features)
+               view-by (om/get-props owner [:map-page :view-by])]
+           (when (pos? no-of-features)
+             (when (= no-of-features 1)
+               (let [feature-id (or
+                                 (-> (first features) (aget "properties")
```

Comment:
this repeats line 400-403, abstract to a function. Also, in that function, only bind `(-> (first features) (aget "properties")` to a var instead of repeating it in each disjunct